azure/main.tf

```
@@ -0,0 +1,161 @@
+locals {
+  rg_ref = var.resource_group_create ? azurerm_resource_group.rg[0].id : data.azurerm_resource_group.rg[0].id
+}
+
+resource "azurerm_resource_provider_registration" "registrations" {
+  for_each = {
+    "Microsoft.Network" = {
+      // Allow deploying V2 application gateways with only a private IP
+      // https://learn.microsoft.com/en-us/azure/application-gateway/application-gateway-private-deployment
+      features = {
+        "EnableApplicationGatewayNetworkIsolation" : true,
+      }
+    },
+    "Microsoft.ContainerService" = {
+      // Use Azure Linux 3.0,
+      // when creating clusters or node pools,
+      // with the AzureLinux OS SKU.
+      features = {
+        "AzureLinuxV3Preview" : true
+      }
```

nit: This was just made GA, so to avoid needing to opt people into this preview flag, wdyt about dropping this and bumping the Kubernetes version to 1.32 instead?

https://techcommunity.microsoft.com/blog/linuxandopensourceblog/azure-linux-3-0-now-generally-available-with-azure-kubernetes-service-v1-32/4399804

```suggestion
      features = {}
```

---

done